google/cloud/storagecontrol/v2/samples/storage_control_folder_samples.cc

```diff
@@ -0,0 +1,219 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storagecontrol/v2/storage_control_client.h"
+#include "google/cloud/internal/getenv.h"
+#include "google/cloud/internal/time_utils.h"
+#include "google/cloud/status.h"
+#include "google/cloud/testing_util/example_driver.h"
+#include <chrono>
+#include <iostream>
+#include <regex>
+#include <string>
+#include <vector>
+
+namespace {
+
+void CreateFolder(google::cloud::storagecontrol_v2::StorageControlClient client,
+                  std::vector<std::string> const& argv) {
+  // [START storage_control_create_folder]
+  namespace storagecontrol = google::cloud::storagecontrol_v2;
+  [](storagecontrol::StorageControlClient client,
+     std::string const& bucket_name, std::string const& folder_id) {
+    auto const parent = std::string{"projects/_/buckets/"} + bucket_name;
+    auto folder = client.CreateFolder(
+        parent, google::storage::control::v2::Folder{}, folder_id);
+    if (!folder) throw std::move(folder).status();
+
+    std::cout << "Created folder: " << folder->name() << "\n";
+  }
+  // [END storage_control_create_folder]
+  (std::move(client), argv.at(0), argv.at(1));
+}
+
+void DeleteFolder(google::cloud::storagecontrol_v2::StorageControlClient client,
+                  std::vector<std::string> const& argv) {
+  // [START storage_control_delete_folder]
+  namespace storagecontrol = google::cloud::storagecontrol_v2;
+  [](storagecontrol::StorageControlClient client,
+     std::string const& bucket_name, std::string const& folder_id) {
+    auto const name = std::string{"projects/_/buckets/"} + bucket_name +
+                      std::string{"/folders/"} + folder_id;
```

Comment:
optional: just say ` + "/folders/"`.  Effectively, once the left side of `+` is a `std::string` you can add literal strings to it.